### PR TITLE
Add rosbag2_py::Player::play to replace rosbag2_transport_python version

### PR DESCRIPTION
**Pull request by emersonknapp**
"record" is an easy followup if we like this pattern.

As I'm preparing to add more playback arguments, I don't want to manage the `PyArg_ParseTupleAndKeywords` monster anymore

### CODE REVIEW

File: ros2bag/ros2bag/verb/play.py [hunk 1]
```diff
@@ -83,20 +86,25 @@ def main(self, *, args):  # noqa: D102
...
+        topic_remapping = ['--ros-args']
+        for remap_rule in args.remap:
+            topic_remapping.append('--remap')
+            topic_remapping.append(remap_rule)
+
+        storage_options = StorageOptions(
             uri=args.bag_file,
             storage_id=args.storage,
-            node_prefix=NODE_NAME_PREFIX,
+            storage_config_uri=storage_config_file,
```
Comment (jaisontj) [inline ros2bag/ros2bag/verb/play.py:97]: Havnt looked into `StorageOptions` but an empty string is alright here, I assume

File: rosbag2_py/src/rosbag2_py/_transport.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+public:
+  Transport() {}
+
+  virtual ~Transport() = default;
+
+  void play(
+    const rosbag2_storage::StorageOptions & storage_options,
+    const rosbag2_transport::PlayOptions & play_options)
+  {
+    printf("Entered rosbag2_py::Transport::play\n");
```
Comment (jaisontj) [inline rosbag2_py/src/rosbag2_py/_transport.cpp:50]: No common logger used instead of `printf`?

**Review by jaisontj — APPROVED** [review]

File: rosbag2_py/src/rosbag2_py/_transport.cpp (hunk 2 above)
Comment (emersonknapp) [reply rosbag2_py/src/rosbag2_py/_transport.cpp:50]: Oops! Leftover from development, will remove, good catch

File: ros2bag/ros2bag/verb/play.py (hunk 1 above)
Comment (emersonknapp) [reply ros2bag/ros2bag/verb/play.py:97]: Yep! Given that the field is a string, there isn't another option for omitting

**Comment by emersonknapp** [comment]
Gist: https://gist.githubusercontent.com/emersonknapp/65fb16be65c57a7a6f0ed2efec5f12a5/raw/8bc9675254d9aa9bbf2a0cc66798aa03e88ea595/ros2.repos
BUILD args: --packages-up-to rosbag2_transport rosbag2_tests ros2bag rosbag2_py
TEST args: --packages-select rosbag2_transport rosbag2_tests ros2bag rosbag2_py
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/8000
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14101/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8914/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11790/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14198/)

**Pull request merged**